pypsa/plot.py

```diff
@@ -247,27 +258,42 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
         bus_collection = ax.scatter(x, y, c=c, s=s, cmap=bus_cmap, edgecolor='face')
 
     def as_branch_series(ser):
+        # ensure that this function always return a multiindexed series
         if isinstance(ser, dict) and set(ser).issubset(branch_components):
-            return pd.Series(ser)
-        elif isinstance(ser, pd.Series):
-            if isinstance(ser.index, pd.MultiIndex):
-                return ser
-            index = ser.index
-            ser = ser.values
+            return pd.concat(
+                    {c.name: pd.Series(s, index=c.df.index) for c, s in
+                         zip(network.iterate_components(ser.keys()), ser.values())},
+                    names=['component', 'name'])
+        elif isinstance(ser, pd.Series) and isinstance(ser.index, pd.MultiIndex):
+                return ser.rename_axis(index=['component', 'name'])
         else:
-            index = network.lines.index
-        return pd.Series(ser,
-                         index=pd.MultiIndex(levels=(["Line"], index),
-                                             codes=(np.zeros(len(index)),
-                                                    np.arange(len(index)))))
+            ser =  pd.Series(ser, network.lines.index)
+        return pd.concat([ser], axis=0, keys=['Line'],
```

Comment:
Please indent to `ser` level, since it's only important for this branch (clarity).